### PR TITLE
Grace Joh

### PR DESCRIPTION
**Pull request by grace-joh**

### CODE REVIEW

File: lib/course.rb [hunk 1]
```diff
@@ -1 +1,19 @@
+class Course
```
Comment (epintozzi) [inline lib/course.rb:1]: 🔥 This is great! Not a thing I'd change in this class.

File: lib/gradebook.rb [hunk 2]
```diff
@@ -1 +1,36 @@
...
+    @instructor = instructor
+    @courses = []
+  end
+
+  def add_course(course)
+    @courses << course
+  end
+
+  def list_all_students
+    return 'No courses have been added.' if @courses.empty?
```
Comment (epintozzi) [inline lib/gradebook.rb:14]: I love that you're doing this check here. This is a great habit to get in to and will serve you well once you start working with Rails.

File: lib/gradebook.rb [hunk 3]
```diff
@@ -1 +1,36 @@
...
+
+    all_students = {}
+    @courses.each do |course|
+      all_students[course.name.to_sym] = course.students
```
Comment (epintozzi) [inline lib/gradebook.rb:18]: The [assessment requirements](https://github.com/turingschool-examples/students_and_courses_2301#iteration-3) ask for the hash key to be a `Course` object, not the course's name. I'm not taking points off for it this time, but be aware of the details being asked of you. This would not be considered correct if it were the final assessment.

File: lib/gradebook.rb [hunk 4]
```diff
@@ -1 +1,36 @@
...
+    end
+    all_students
+  end
+
+  def students_below(threshold)
+    return 'No courses have been added.' if @courses.empty?
+    return 'No students are currently enrolled.' if self.list_all_students.values.flatten.empty?
```
Comment (epintozzi) [inline lib/gradebook.rb:25]: This is great!

File: lib/university.rb [hunk 5]
```diff
@@ -0,0 +1,17 @@
+# University has name and array of all gradebooks
+# add_gradebook add a gradebook that has instructor and course
+# instructors method tracks all instructors from every gradebook
+# all_grades returns a hash
+# all_grades = { instructor : instructor_name
+#                course     : course_name
+#                grade      : [array of all student averages]
+# students_within(lower_bound, higher_bound)
```
Comment (epintozzi) [inline lib/university.rb:8]: This is a great starting point and really smart to layout what you want your method to do before you start writing it.

File: spec/course_spec.rb [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+      @course1.enroll(@student2)
+
+      expect(@course1.students).to eq([@student1, @student2])
+    end
+
+    it 'can alert that a course is full and will not add students' do
+      @course1.enroll(@student1)
+      @course1.enroll(@student2)
+
+      expect(@course1.enroll(@student3)).to eq('The course is full.')
```
Comment (epintozzi) [inline spec/course_spec.rb:53]: Excellent! I'm happy to see you testing all possible outcomes for this method.

**Review by epintozzi — COMMENTED** [review]
Excellent work! I love all the edge cases you've covered both in your tests and in the code.
